lib/ansible/modules/network/vyos/vyos_lldp_global.py

```diff
@@ -0,0 +1,318 @@
+#        "delete service lldp legacy-protocols fdp",
+#        "set service lldp management-address '192.0.2.14'",
+#        "set service lldp legacy-protocols edp",
+#        "set service lldp legacy-protocols sonmp"
+#    ]
+#
+# "after": [
+#        {
+#            "address": "192.0.2.14"
+#        },
+#        {
+#            "legacy_protocols": [
+#                "cdp",
+#                "edp",
+#                "sonmp"
+#            ]
+#        }
+#        {
+#            "enable": true
+#        }
+#    ]
+#
+# After state:
+# -------------
+#
+# vyos@vyos:~$ show configuration commands|grep lldp
+# set service lldp legacy-protocols cdp
+# set service lldp legacy-protocols edp
+# set service lldp legacy-protocols sonmp
+# set service lldp management-address '192.0.2.14'
+
+
+# Using deleted
+#
+# Before state
+# -------------
+# vyos@vyos:~$ show configuration commands|grep lldp
+# set service lldp legacy-protocols cdp
+# set service lldp legacy-protocols edp
+# set service lldp legacy-protocols sonmp
+# set service lldp management-address '192.0.2.14'
+#
+- name: Delete attributes of given lldp service (This won't delete the LLDP service itself)
+  vyos_lldp_global:
+    config:
+    state: deleted
+#
+#
+# ------------------------
+# Module Execution Results
+# ------------------------
+#
+# "before": [
+#        {
+#            "address": "192.0.2.14"
+#        },
+#        {
+#            "legacy_protocols": [
+#                "cdp",
+#                "edp",
+#                "sonmp"
+#            ]
+#        }
+#        {
+#            "enable": true
+#        }
+#    ]
+#
+#  "commands": [
+#       "delete service lldp management-address",
+#        "delete service lldp legacy-protocols"
+#    ]
+#
+# "after": [
+#        {
+#            "enable": true
+#        }
+#          ]
+#
+# After state
+# ------------
+# vyos@vyos:~$ show configuration commands | grep lldp
+# set service lldp
+
+
+"""
+RETURN = """
+before:
+  description: The configuration prior to the model invocation.
+  returned: always
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+after:
+  description: The resulting configuration model invocation.
+  returned: when changed
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample:
+    - set service lldp legacy-protocols sonmp
+    - set service lldp management-address '192.0.2.14'
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.vyos.argspec.lldp_global.lldp_global import Lldp_globalArgs
+from ansible.module_utils.network.vyos.config.lldp_global.lldp_global import Lldp_global
+
+
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
```

Comment:
We need a `required_if` to fail early if no config is provided but state is either merged or replaced.

Reply:
done